src/cdk/tree/tree.ts

```diff
@@ -268,7 +272,23 @@ export class CdkTree<T, K = T>
   private _viewInit = false;
 
   constructor(...args: unknown[]);
-  constructor() {}
+  constructor() {
+    effect(
+      onCleanup => {
+        const data = this._data();
+        const nodeType = this._nodeType();
+        const expandedKeys = this._selection();
+
+        const sub = this._getRenderData(data, nodeType, expandedKeys).subscribe(renderData => {
+          this._renderDataChanges(renderData);
+        });
+        onCleanup(() => {
+          sub.unsubscribe();
+        });
+      },
+      {allowSignalWrites: true},
```

Comment:
Could we use `computed` instead of writing inside of an effect?

Reply:
I tried to reduce the complexity of the effect, but I think a lot more of this component would need to be restructured to make that possible.

Reply:
We're not on that version of Angular yet, but I'd be a bit concerned if things will break once we update to a version that has https://github.com/angular/angular/pull/57874.

Reply:
let me take another stab at this